### ISSUE TITLE
linux : crash with bsfnullrenderapi.so

### ISSUE THREAD

**Issue by XMatrixXiang** [GameFoundry/bsf#345]
i build bsf with this  command:
cmake ../ -D RENDERER_MODULE=Null -D RENDER_API_MODULE=Null -D CMAKE_BUILD_TYPE=Debug -D AUDIO_MODULE=Null -D PHYSICS_MODULE=Null

**Comment by pgruenbacher** [comment]
was this on linux? can you provide an example case to replicate?

**Comment by XMatrixXiang** [comment]
@pgruenbacher  yes it was crashed on linux! By this way you can reproduce this bug!



`
#include "BsApplication.h"

namespace bs
{
	UINT32 windowResWidth = 1280;
	UINT32 windowResHeight = 720;
}


/** Main entry point into the application. */
#if BS_PLATFORM == BS_PLATFORM_WIN32

int CALLBACK WinMain(
	_In_  HINSTANCE hInstance,
	_In_  HINSTANCE hPrevInstance,
	_In_  LPSTR lpCmdLine,
	_In_  int nCmdShow
	)
#else
int main()
#endif
{
	using namespace bs;

	// Initializes the application and creates a window with the specified properties
	VideoMode videoMode(windowResWidth, windowResHeight);
	Application::startUp(videoMode, "DebugWindow", false);


	// Runs the main loop that does most of the work. This method will exit when user closes the main
	// window or exits in some other way.
	Application::instance().runMainLoop();

	CppHttpLibServerConnector::shutDown();
	// When done, clean up
	Application::shutDown();

	return 0;
}`
